external/anomaly/ote_anomalib/train_task.py

```diff
@@ -0,0 +1,88 @@
+"""Anomaly Classification Task."""
+
+# Copyright (C) 2021 Intel Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions
+# and limitations under the License.
+
+import io
+
+import torch
+from anomalib.utils.callbacks import MinMaxNormalizationCallback
+from ote_anomalib import AnomalyInferenceTask
+from ote_anomalib.callbacks import ProgressCallback
+from ote_anomalib.data import OTEAnomalyDataModule
+from ote_anomalib.logging import get_logger
+from ote_sdk.entities.datasets import DatasetEntity
+from ote_sdk.entities.metrics import Performance, ScoreMetric
+from ote_sdk.entities.model import ModelEntity, ModelPrecision
+from ote_sdk.entities.train_parameters import TrainParameters
+from ote_sdk.serialization.label_mapper import label_schema_to_bytes
+from ote_sdk.usecases.tasks.interfaces.training_interface import ITrainingTask
+from pytorch_lightning import Trainer
+
+logger = get_logger(__name__)
+
+
+class AnomalyTrainingTask(AnomalyInferenceTask, ITrainingTask):
```

Comment:
As mentioned above, I think this class should be BaseAnomalyTask. `AnomalyTrainingTask` task doesn't reveal the class' intention. The class inherits both Training and Inference Tasks, and is named as `AnomalyTrainingTask`. Based on the entrypoints in model templates, this behaviour is equivalent to base training/inference. That's why I would still name this class as `BaseAnomalyTask`, and named the module as `external/anomaly/ote_anomalib/base_task.py`